src/reagent/ratom.cljs

```diff
@@ -456,19 +456,17 @@
   (-deref [this]
     (when-some [e caught]
       (throw e))
-    (let [non-reactive (nil? *ratom-context*)]
-      (when non-reactive
-        (flush!))
-      (if (and non-reactive (nil? auto-run))
-        (when dirty?
-          (let [oldstate state]
-            (set! state (f))
-            (when-not (or (nil? watches) (= oldstate state))
-              (notify-w this oldstate state))))
-        (do
-          (notify-deref-watcher! this)
-          (when dirty?
-            (._run this false)))))
+    (if (reactive?)
+      (notify-deref-watcher! this)
```

Comment:
Previously the branch where `notify-deref-watcher!` is called was also run if `(nil? auto-run)` was falsey.

```
(and non-reactive (nil? auto-run))
(or (not non-reactive) (not (nil? auto-run)))
(or reactive (some? auto-run))
```

Tests seem to pass, so not sure if this is important. Or this case may be missing a test.